gradle-plugin/src/main/kotlin/arrow/meta/plugin/gradle/ArrowKotlinGradleSubplugin.kt

```diff
@@ -13,7 +13,7 @@ import org.jetbrains.kotlin.gradle.plugin.SubpluginOption
  */
 class ArrowKotlinGradleSubplugin : KotlinGradleSubplugin<AbstractCompile> {
   companion object {
-    private const val GRADLE_ARTIFACT_NAME = "gradle-plugin"
+    private const val KOTLIN_ARTIFACT_NAME = "compiler-plugin"
```

Comment:
You should not change the names here

Reply:
For me `GRADLE_ARTIFACT_NAME` means gradle plugin, not kotlin compiler plugin that it's supposed to be

Reply:
That is not the intent what you might think.
We supply tooling in the idea plugin to get the gradle-plugin, which does the registration automatically, based on rules we define in `ArrowGradlePlugin`. Because you deleted it it makes sense just to retrieve it directly. 
With the changes, you introduce your naming the gradle-plugin `compiler-plugin`, which doesn't make sense to me.